### PR TITLE
otelzap: Add skeleton and config options 

### PR DESCRIPTION
**Pull request by khushijain21**
Part of  https://github.com/open-telemetry/opentelemetry-go-contrib/issues/5191

Pre-work https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5279

### CODE REVIEW

**Comment by pellared** [comment]
Can you please outline an implementation plan in https://github.com/open-telemetry/opentelemetry-go-contrib/issues/5191?

File: bridges/otelzap/config.go
Comment (pellared) [inline bridges/otelzap/config.go:4]: Have you added ` // import "otelzap"` manually? The import path should be `go.opentelemetry.io/contrib/bridges/otelzap` and should be generated automatically by `make`.

File: bridges/otelzap/config.go
Comment (pellared) [inline bridges/otelzap/config.go:49]:
```suggestion
// Option configures a [Core].
```

File: bridges/otelzap/config.go
Comment (pellared) [inline bridges/otelzap/config.go:59]:
```suggestion
// the [log.Logger] used by a [Core].
```

File: bridges/otelzap/config.go
Comment (pellared) [inline bridges/otelzap/config.go:61]:
```suggestion
// By default if this Option is not provided, [Core] will use a default
```

File: bridges/otelzap/core.go
Comment (pellared) [inline bridges/otelzap/core.go:27]:
```suggestion
// NewCore creates a new [zapcore.Core] that can be used with [go.uber.org/zap.New].
```

File: bridges/otelzap/core.go
Comment (pellared) [inline bridges/otelzap/core.go:47]: Add a TODO comment as I think Sync will be a no-op.

File: bridges/otelzap/core_test.go
Comment (pellared) [inline bridges/otelzap/core_test.go:4]: Is there anything taken from Uber code here?

```suggestion
```

File: versions.yaml
Comment (pellared) [inline versions.yaml:84]: Can you reorder (alphabetical) and add an EOL?

File: bridges/otelzap/core_test.go
Comment (pellared) [inline bridges/otelzap/core_test.go:48]:
```suggestion
		l := h.logger.(*logtest.Recorder)
		require.Len(t, l.Result(), 1)

		want := &logtest.ScopeRecords{Name: scope.Name, Version: scope.Version, SchemaURL: scope.SchemaURL}
		got := l.Result()[0]
		assert.Equal(t, want, got)
```

File: bridges/otelzap/core_test.go
Comment (pellared) [inline bridges/otelzap/core_test.go:29]:
```suggestion
		require.NotPanics(t, func() { h = NewCore() })
		require.NotNil(t, h.logger)
		require.IsType(t, &logtest.Recorder{}, h.logger)
		l := h.logger.(*logtest.Recorder)
		require.Len(t, l.Result(), 1)
		
		want := &logtest.ScopeRecords{Name: bridgeName, Version: version}
		got := l.Result()[0]
		assert.Equal(t, want, got)
```

**Comment by pellared** [comment]
Can you also update the `CODEOWNERS` file? You can add me as one of the codeowners.

Do you want to be a codeowner as well? Will you be able to maintain otelzap after your internship is finished?

**Comment by khushijain21** [comment]
yes! would like to maintain it after the internship too

**Review by MrAlias — APPROVED** [review]

**Review by pellared — APPROVED** [review]

**Comment by pellared** [comment]
@open-telemetry/go-approvers, PTAL. This PR needs a second approval from a non-Splunk employee.

**Review by dashpole — APPROVED** [review]

**Pull request merged**
